### ISSUE TITLE
SNMessenger does work on Sideloaded Devices

### ISSUE THREAD

**Issue by iammanpreetsingh** [NguyenASang/SNMessenger#4]
> Currently, the iPA doesn't work on non-jailbroken devices and cannot be sideloaded. I will try to fix it as soon as possible.

As stated in your release information.

But it does work 100% fine. We know how to made it work. If you need help, you can come to [AppTesters.org](https://apptesters.org) or telegram: @[AppleTesters ](https://t.me/AppleTesters)

https://streamable.com/m6vkgf (Video will expire in 2 days)

**Comment by NguyenASang** [comment]
@iammanpreetsingh Yes, it will work if we somehow preserve the app's identifier. If you use Sideloadly with a regular Apple account, it will use an `automatic bundle ID` , which causes problems.

**Comment by NguyenASang** [comment]
Closed due to inactivity.

**Comment by iammanpreetsingh** [comment]
Could you kindly elaborate on what is not working? I do not use Messenger frequently, so I may have overlooked something.

**Comment by NguyenASang** [comment]
Every feature that hooks functions (using `%hookf`) such as: "disable read receipts", "audio / video call confirmation",... doesn't work correctly on non-jailbroken devices. Another alternative for non-jailbroken is `Fishhook`, but it doesn't work with private functions.

And yes, I can create an iPA for a non-jailbroken device, but it will lack some features (that's why I dropped support for it until I found a way to solve that problem).

**Comment by KTsarlz** [comment]
@NguyenASang Is it really not working still for the disable read receipts?
